packages/widget/src/components/transfer/fungible/fungible-token-transfer.ts

```diff
@@ -58,15 +59,19 @@ export class FungibleTokenTransfer extends BaseComponent {
         break;
       case FungibleTransferState.WRONG_CHAIN:
         {
-          this.walletController.switchChain(
-            this.transferController.sourceNetwork!.chainId
+          void this.walletController.switchEvmChain(
+            this.transferController.sourceNetwork!.chainId,
+            this.transferController.walletContext.value?.evmWallet
+              ?.provider as Eip1193Provider
           );
         }
         break;
       case FungibleTransferState.COMPLETED:
         {
-          this.walletController.switchChain(
-            this.transferController.sourceNetwork!.chainId
+          void this.walletController.switchEvmChain(
```

Comment:
@wainola may i know why switching happens when a transfer is completed? just need context

Reply:
honestly I don't know, and removing this doesn't seem to have any effect when trying to make another transfer, once the first one was finished. @mpetrunic was there any special reason to call the method here once the transfer was complete?

Reply:
Pretty sure it didnt. Its probably some bad merge?!